src/events/messageReactionAdd.ts

```diff
@@ -37,68 +41,30 @@ export default async (client: Discord.Client, reaction: Discord.MessageReaction,
-        const veteranVotes = { up: project.upvotes.veterans, down: project.downvotes.veterans }
-
-        log.info(`Project ${project.name} (ID ${project.id}) was ${wasApproved ? 'approved' : 'rejected'} with ${staffVotes.up + veteranVotes.up} upvotes [Staff/vet spread: ${staffVotes.up} | ${veteranVotes.up}] and ${staffVotes.down + veteranVotes.down} downvotes [Staff/vet spread: ${staffVotes.down} | ${veteranVotes.down}]`)
-
-        const voteSituation = `**Upvotes:** **${project.upvotes.staff}** staff, **${project.upvotes.veterans}** veterans\n**Downvotes:** **${project.downvotes.staff}** staff, **${project.downvotes.veterans}** veterans`
-        await safeSendMessage(channel, `${wasApproved ? '✅' : '❌'} Project **${project.name}** (${project.links.source}, ID ${project.id}) was **${wasApproved ? 'APPROVED' : 'REJECTED'}** by **${user.tag}** (${user.id}) with following vote situation:\n${voteSituation}`)
-
-        try {
-          await reaction.message.delete({ reason: `Project ${wasApproved ? 'approved' : 'rejected'} by ${user.tag} (${user.id})` })
-        } catch (err) {
-          log.error(`Could not delete submission post for ${project.name} (ID ${project.id}): ${err}`)
-          await safeSendMessage(channel, `⚠️ Could not delete project submission post. Please delete message ${project.id} manually. (Discord error)`)
-        }
+      const input: ShowcaseInput = {
+        result,
+        isPause
       }
 
-      // Post to public showcase
-      if (wasApproved) {
-        try {
-          if (!process.env.PROJECT_SHOWCASE_CHANNEL) {
-            throw new Error(`Project showcase channel ID not set, got ${process.env.PROJECT_SHOWCASE_CHANNEL}`)
-          }
-
-          // Having to type cast here and just separately check that this channel has a send method
-          // FWIW, this seems to indeed be the official recommended method by the discord.js development team: https://github.com/discordjs/discord.js/issues/3622#issuecomment-565566337
-          const showcaseChannel = guild.channels.cache.get(process.env.PROJECT_SHOWCASE_CHANNEL) as TextChannel
-
-          if (!showcaseChannel || !showcaseChannel.send) {
-            throw new Error('Project showcase channel not found in cache or is not a text channel, possible configuration error')
-          }
-
-          const embed = createProjectEmbed(project, guild)
-          await showcaseChannel.send(null, embed)
-          log.info(`Project ${project.name} (ID ${project.id}) posted to showcase channel.`)
-        } catch (err) {
-          log.error(`Could not post project ${project.name} (ID ${project.id}) to showcase channel: ${err}`)
-          await safeSendMessage(channel, '⚠️ Could not post project to showcase channel. (Internal error)')
-        }
+      const dInput: ShowcaseDiscordInput = {
+        guild,
+        channel,
+        user,
+        reaction
       }
+      await showcase(dInput, input)
```

Comment:
Minor nit: We're not in GoLand, so let's use descriptive variable names please.

Another thing: You don't appear to be catching errors you throw in the `showcase` function. You'll want to catch and log that.

```suggestion
      const discordInput: ShowcaseDiscordInput = {
        guild,
        channel,
        user,
        reaction
      }
      await showcase(discordInput, input)
```